By popular demand, nerfs mining

Rukofamicom opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
### It nerfs mining. Again. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
### Single most requested change for the game on the Beestation forums.

Memes aside:
#### section deleted while I'm still working on making this an actual good PR so people stop accusing what was part of the meme as being a finalized version. Go read the god damn code or shut up about something I've already removed being bad while I'm working. 

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->

Tested legion cores on both ashwalkers and unbelievers
Tested ash storms on ashwalker
Checked mining vendor to ensure the new asteroid kit dispenses
Tested out the new hiero chasers with crusher


https://user-images.githubusercontent.com/9547572/166311713-0e5fca18-9489-4e0f-afe9-224b7f4046c9.mp4

![image](https://user-images.githubusercontent.com/9547572/166311503-c60d0276-87d6-4bfe-8035-ec583aa0509e.png)
![image](https://user-images.githubusercontent.com/9547572/166311649-9922fd44-d2c1-4b5f-9571-af4219dff753.png)


## Changelog
:cl:
to be filled out when finalized. 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rukofamicom committed:
By popular demand, nerf's mining

Rukofamicom committed:
Update machine_vending.dm

Rukofamicom committed:
Heiro trophy wall is solid

Rukofamicom committed:
buffs mining??

Rukofamicom committed:
nerfs miners, buffs ashwalkers

Rukofamicom committed:
lowmem whoops

Liidero commented:
I like mining and I like the sound of these changes

EvilDragonfiend commented on code/modules/exploration_crew/exploration_vendor.dm:
```diff
@@ -14,9 +14,6 @@ GLOBAL_VAR_INIT(exploration_points, 0)
 		new /datum/data/vendor_equipment("Survival Medipen",			/obj/item/reagent_containers/hypospray/medipen/survival,			2000),
-		new /datum/data/vendor_equipment("Brute Healing Kit",			/obj/item/storage/firstaid/brute,									3000),
-		new /datum/data/vendor_equipment("Burn Healing Kit",			/obj/item/storage/firstaid/fire,									3000),
-		new /datum/data/vendor_equipment("Advanced Healing Kit",		/obj/item/storage/firstaid/advanced,								5000),
```
exploration crews are already limited to healing themselves as their points are not that very rich as much as miner's. I believe you don't have to touch these.
I think you thought these are for miners.

Rukofamicom replied:
It was 100% intentional.

> Any role being 100% self-reliant is bad.

lowwy commented on code/modules/mining/machine_vending.dm:
```diff
@@ -134,18 +134,15 @@
-		new /datum/data/vendor_equipment("Survival Medipen",			/obj/item/reagent_containers/hypospray/medipen/survival,			500),
-		new /datum/data/vendor_equipment("Brute Healing Kit",			/obj/item/storage/firstaid/brute,									600),
-		new /datum/data/vendor_equipment("Burn Healing Kit",			/obj/item/storage/firstaid/fire,									600),
-		new /datum/data/vendor_equipment("Advanced Healing Kit",		/obj/item/storage/firstaid/advanced,								1200),
```
I'm scared that besides explorers, miners will also head to medbay to raid medkits with this gone.

Rukofamicom replied:
I don't appreciate the survival pen being highlighted with the others as if it is removed. The survival pen is still present, just with an adjusted cost. 

Again:
> Any role being 100% self-reliant is bad.

Injured players should rely on medbay for treatment.

ItsTisme commented:
maby making the walls manualy dispellable, otherwise the hiero trophy is arguebly a downgrade since it would also block youre own movement. And I dont think you can remove trophys, you could very easly get yourself stuck and dragons flames still pierce it

imsynthex commented:
maybe make hiero walls work like wizard forcewall in that you get to spawn them in at will by pressing a button

RDS88-beeman commented:
> since i don't think you can remove trophys

crowbar works

Rukofamicom commented:
What RDS said. 
I could make it last for less time though so it's less obstructive if there's support for that.

williamcll commented:
How about instead of regen core buff, Ashwalkers are immune to ash storms?
Would fit the name better, and there ARE legion ash walkers so it fits the theme more.

Also would be nice to follow corgstation and have roundstart spacesuit/Hardsuit for the low risk low reward asteroid mining.

Rukofamicom commented:
> How about instead of regen core buff, Ashwalkers are immune to ash storms?

Because this means being totally heatproof, which tbh I don't see an issue with doing. I do *also* think they deserve the legion core thing. They have no means of stabilizing legion cores, so outside of getting a lucky tendril spawn they have no possible way to stockpile them.

> Also would be nice to follow corgstation and have roundstart spacesuit/Hardsuit for the low risk low reward asteroid mining.

I will add it to the voucher machine as one of the starting options alongside a GPS and full size O2 tank, because I agree with asteroid mining being a viable option to choose.

Rukofamicom committed:
Buffs miners??? Is this the right PR?

Rukofamicom commented:
There we go:

* Hiero staff no longer has a wall at all, instead firing an offensive hiero projectile on detonation. 
* Ashwalkers immune to Ash storm
* Mining vendor has a voucher option for asteroid mining kit.

civilCornball commented:
I hate how much of a mixed bag this one is.
There’s so much good stuff, but everything else I dislike.
In order:
-(ore satchel storage) Unsure if I like or dislike this. Always thought ore satchel size was great, and encouraging ppl to take ore boxes means they know they have portable cover.
-Survival medipens… why? I generally hover around 2K points.
-Skeleton keys… I understand this, but most tendril loot is worth about 750 points. **Better idea**: boss chest specific key that costs 6k. It’ll get the boss rushers to actually send ore back up.
-Medkits removed… I understand why you want this, but I’m worried that it’ll hurt newer players that can’t navigate well enough to find medbay.
Also, what if they’re bleeding and don’t know that their capsule has bandages in the portable medbay thing?
-Heiro det now creates chaser… hek yes
-Legion core nerf (and yes, it IS a nerf)… why? At least make it cause the player to regenerate over time, to complement the instant heal of the survivor kit.
-Ashwalker buff… HELL YES.
-Asteroid mining… Fun idea, but might cause problems when more than one person wants the mining shuttle.

As an aside, while we’re changing mining: can you lower the prices on the more expensive emergency capsules? I never see them, and I’m never able to afford them. (This lack of a pointsink may be contributing to mining sending out piles and piles of normally mining-specific supplies. Like those medkits you’re complaining about!)

Rukofamicom commented:
> Survival medipens… why?
Because they border on being a full heal over time with a pretty short cooldown between uses, despite implying you need to wait a long time.

Contents:
> Salbutamol, 100 ticks worth, 300 total oxyloss healing (debatably negligible despite the high number)
> Leporazine, 100 ticks worth, makes you extremely heat resistant in case of fires
> Tricordrazine, 38 ticks worth, 30 brute/burn/tox/oxy healing
> Epinephrine, 100 ticks worth, near immunity to stuns (such as goliath tendril)
> Lavaland extract, 5 ticks worth, 20 brute/burn, 15 tox healing
> Omnizine, 50 ticks worth, 25 brute/burn/tox/oxy
> **Total healing over 50 ticks: 75 brute/burn, 70 tox,**

They're pretty damn good at being the only med you you'll need unless you're being greedy/aggressive. And readily available healing for mining is not nearly as mandatory as it's often made out to be, unless this is the case.  If someone sees a tendril when they don't have healing support items *they can go the other way and come back later*. Nothing except greed is compelling them to fight a watcher tendril while they're underprepared. 

There's also the legion cores for an extra +30 healing every few minutes, and then lavaland cactus as well which have healing properties... if you don't overdo them and get fat. 

> -Medkits removed… I understand why you want this, but I’m worried that it’ll hurt newer players that can’t navigate well enough to find medbay.

New players that cannot find the medbay are going to have far more issues than missing medkits. Newer miners also tend to avoid combat at first. 

> -Legion core nerf (and yes, it IS a nerf)… why? At least make it cause the player to regenerate over time, to complement the instant heal of the survivor kit.

because a frequently dropped instant full heal is ridiculous. Even if I made legion cores work over time, if I kept them as full heals they'd be the only thing miners ever need to survive. *Hell they currently are the only thing most miners I see need to survive, no matter how aggressively they play*.

> -Asteroid mining… Fun idea, but might cause problems when more than one person wants the mining shuttle.

Already does. I've just streamlined the ability to choose asteroid mining, which is mostly safe and devoid of both monsters and lava. You'll want to pick up a fulton kit ASAP though so you aren't relying on co-workers to pick you up when you have a load to bring back.

Rukofamicom committed:
discovery scanners

Rukofamicom commented:
Alright, I didn't even try to hide that this started as a meme, but I'm actually going to try to make something good out of this. 

There will be nerfs to it, but fucking over good miners just because of some bad miners isn't the right way to go and I'm worried this might actually have a chance of getting merged in its current state. I do hold a lot of the beliefs I've stated, but not to the extreme I have represented so far. 

let me roll back the memes a bit.

EvilDragonfiend commented on code/game/objects/items/storage/bags.dm:
```diff
@@ -133,7 +133,7 @@
 	STR.can_hold = typecacheof(list(/obj/item/stack/ore))
 	STR.max_w_class = WEIGHT_CLASS_HUGE
-	STR.max_combined_stack_amount = 50
+	STR.max_combined_stack_amount = 400
```
I believe this isn't `to 350`, this is `increased by 350`

Rukofamicom replied:
Yes, I believe it *should* have been 400, but testing revealed it to be 350. I'm not sure why, but that's how it be. 

![image](https://user-images.githubusercontent.com/9547572/166400414-79762883-6628-4455-8001-e3c2417c376f.png)

![image](https://user-images.githubusercontent.com/9547572/166400426-f4980812-848c-49e6-88e6-486a4409f093.png)

EvilDragonfiend replied:
I believe it is because `max_items = 7` is declared in `/datum/component/storage`, which is 7 stacks. you need to make set it 8.

EvilDragonfiend replied:
hmm, I think it means as well you need to set `max_items` as higher value more than just 8 if you want multiple types of ore in a bag as there are 11 types of ore. in my calculation, you should set it 20.(in case you put weird stacks of ore regardless of their types)

Rukofamicom replied:
Sounds good, and I've had sleep now too. 
I'll turn the max items up and adjust the total to 350

Rukofamicom committed:
assorted changes

TODO:
Boss drops in locked chests now

Replace the puzzle chest with a random, extra boss drop

General debugging and testing.

williamcll commented:
> Alright, I didn't even try to hide that this started as a meme, but I'm actually going to try to make something good out of this.
> 
> There will be nerfs to it, but fucking over good miners just because of some bad miners isn't the right way to go and I'm worried this might actually have a chance of getting merged in its current state. I do hold a lot of the beliefs I've stated, but not to the extreme I have represented so far.
> 
> let me roll back the memes a bit.


What is the ratio between good miners and bad miners anyway? Seems like nerfing shouldn't be done by looking what's on paper but rather how OP mining is by statistics such as how well antag Miners play and/or Miner K/D against other players.

Isy232 commented:
I'm definately not a mining main but I like the sounds of these changes a lot

Buffing Ashwalkers and nerfing miners healing is definately nice.

Rukofamicom committed:
Drops medibot idea

Regenerative core idea better.

Rukofamicom commented:
> What is the ratio between good miners and bad miners anyway? Seems like nerfing shouldn't be done by looking what's on paper but rather how OP mining is by statistics such as how well antag Miners play and/or Miner K/D against other players.

I don't think the ratio matters a lot, but it's almost hardcapped at a third with the miners who are rushing bosses not understanding that "well there are other miners and only one can rush bosses anyway" are being problematic.

Ignore my past comments on the matter about forcing miners to medbay, because these aren't really true and was part of the trolling behind the original PR: I removed the medkits because they're absurdly good *for the station* (especially advanced, which were removed from even medbay ages ago) but serve little purpose for miners in practice because they can't be used in combat and there's plenty of healing available without carrying a bulky medkit. It's rare that I see a miner buying them at all, and more than half of the time I do it's the miner tossing them at someone on the station for use because they were a good miner and had spare points. 

Healing still available to miners without medkits in vendor:
* Legion cores, which are now nerfed unless stabilized (stabilization not pushed yet)
* Survival Pens, which heal ~75% of damage over a short time and provide near immunity to stuns for three minutes
* Lavaland cactus fruit, which are minor heals and will make you fat if overeaten
* All varieties of shelter capsule come with several donk pockets, which again can make you fat if overeaten
* Luxury shelter capsule still comes with one brute and one burn medkit for 3000 points, as well as some RNG food.
* Mining outpost capsule comes with a medbay sleeper pod in the 5000 point mining outpost capsule. This is infinite but stationary and requires a second miner
* Luxury Bar capsule has... well a fully stocked bar which can provide healing if you know how to mix drinks. This one is pretty exorbitantly priced at 10,000 though.

Out of all of those, I'd only say the Bar isn't appropriately priced for the use it provides, but its purpose also isn't healing so much as it is to show off. There's still loads of options that better suit miners despite how I brashly presented the change. If all of those aren't enough, I do realistically think miners should lean on other departments at this point - if not leaning on medbay, ask robotics for a medibot - they're fragile and can't survive an ash storm, but can be left in any shelter pod to return to for healing.

Oscari-pog commented:
I gotta ask "why" because none of this seems necessary. Mining is not rolling over antags, and they got so many nice things taken away already.

This ultimately feels somewhat spiteful.

Rukofamicom committed:
Removes most modkits from tendril table

Buffs lifesteal pka mod to not take up any capacity. It can be stacked, but it can not be reliably farmed.

Rukofamicom commented:
> I gotta ask "why"

Because miners are still
 * Completely ignoring their responsibility to mine ores
 * Living entirely off of the land instead of having a reason to acquire points
 * Relying entirely on research to provide them gear, and then ignoring everything after a scientist expecting ores in exchange pushes a button for plasma cutter

 The changes listed currently are not accurate of what the PR is going to be. I made it a draft because I'm redoing it all. 
 
 The primary focus of the PR as far as nerfs is going to be:
  * almost EVERYTHING progression related will cost points
  * Boss drops will now be in chests that also require keys, like tendrils
  * Powerful (but not all) Healing will be behind points or else relying on station, avoid combat until you can afford it which is quite easy to do. 
  
  Everything else I'm working on will be direct improvements, such as improving tendril loot drops

Rukofamicom commented:
I'm going to break this apart into several smaller PRs so that the good and bad together isn't such a package deal and the powers that be can better choose what they do and don't want to add.
